File: front/api/product.ts
import axios from "axios";

// 상품 등록 요청
export interface ProductCreateRequest {
  name: string;
  price: number;
  description?: string;
  imageUrl?: string;
}

export const createProduct = async (data: ProductCreateRequest) => {
  const response = await axios.post("/api/products", data);
  return response.data;
};

File: back/minishop/minishop/src/main/java/com/example/minishop/entity/Product.java
package com.example.minishop.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@Table(name = "products")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private int price;

    @Column(nullable = false)
    private int stockQuantity;

    @Column(length = 1000)
    private String description;

    private String imageUrl;

    public void update(String name, int price, int stockQuantity, String description, String imageUrl) {
        this.name = name;
        this.price = price;
        this.stockQuantity = stockQuantity;
        this.description = description;
        this.imageUrl = imageUrl;
    }
}
